tests/nnstreamer_filter_single/unittest_filter_single.cc

```diff
@@ -13,212 +13,196 @@
 
 #include "../gst/nnstreamer/tensor_filter/tensor_filter_single.h"
 
+#ifdef ENABLE_TENSORFLOW_LITE
 /**
- * @brief Internal data structure to run tensor-filter single.
+ * @brief Test Fixture class for a tensor-filter single functionality.
  */
-typedef struct
+class NNSFilterSingleTest : public::testing::Test
 {
+protected:
   GTensorFilterSingle *single;
   GTensorFilterSingleClass *klass;
   GstTensorMemory input;
   GstTensorMemory output;
-} single_test_data_s;
-
-#ifdef ENABLE_TENSORFLOW_LITE
-/**
- * @brief Get max score from output.
- */
-static gsize
-get_max_score (GstTensorMemory * output)
-{
-  guint8 *array;
-  gsize idx, max_idx = 0;
-  guint8 max_value = 0;
-
-  array = (guint8 *) output->data;
-
-  for (idx = 0; idx < output->size; idx++) {
-    if (max_value < array[idx]) {
-      max_idx = idx;
-      max_value = array[idx];
-    }
+  gboolean loaded;
+
+public:
+  /**
+   * @brief Construct a new NNSFilterSingleTest object
+   */
+  NNSFilterSingleTest ()
+    : single (nullptr), klass (nullptr), loaded (FALSE)
+  {
+    input.data = nullptr;
+    output.data = nullptr;
   }
 
-  return max_idx;
-}
+  /**
+   * @brief SetUp method for each test case
+   */
+  void SetUp () override
+  {
+    g_autofree gchar *model_file = nullptr;
+    g_autofree gchar *data_file = nullptr;
+    gsize length;
+    const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+
+    input.size = 3U * 224 * 224;
+    output.size = 1001U;
+
+    single = (GTensorFilterSingle *) g_object_new (G_TYPE_TENSOR_FILTER_SINGLE, NULL);
+    klass = (GTensorFilterSingleClass *) g_type_class_ref (G_TYPE_TENSOR_FILTER_SINGLE);
+
+    /* supposed to run test in build directory */
+    if (root_path == NULL)
+      root_path = "..";
+
+    model_file = g_build_filename (root_path, "tests", "test_models", "models",
+        "mobilenet_v1_1.0_224_quant.tflite", NULL);
+    if (!g_file_test (model_file, G_FILE_TEST_EXISTS)) {
```

Comment:
When model file is not in model directory or reading file content fails (line 62 and 68), all testcase should be stopped. 
How about using ASSERT_TRUE() for this case?

Reply:
Good advice! I updated the code as below.
```cpp
ASSERT_TRUE (g_file_test (model_file, G_FILE_TEST_EXISTS));
```